dashboard/app/controllers/projects_controller.rb

```diff
@@ -571,6 +571,39 @@ def limited_gallery?
     !project_validator && limited_project_gallery
   end
 
+  # GET /projects/:channel_id/extra_links
+  # Get the extra links for the project for use by project validators.
+  # This is used by lab2 levels that cannot use the haml 'extra links' box since
+  # this box will not refresh when changing levels.
+  def extra_links
```

Comment:
how are permissions handled for this method?

Reply:
This has been a question that's been floating around in my head so thanks for asking it!
I see that currently, a signed out user doesn't have access to this method but signed-in users do even though the information is not actually displayed via Extra Links.
I'll update and add a condition at the beginning of the definition to check for `project_validator` permission. Thanks!